plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -212,6 +212,21 @@ public PluginCompatReport testPlugins()
         mconfig.userProperties.put( "failIfNoTests", "false" );
         mconfig.userProperties.putAll(this.config.retrieveMavenProperties());
 
+        // Override JDK if passed explicitly
+        if (config.getTestJDKHome() != null) {
+            //TODO: move up
+            File jdkHome = config.getTestJDKHome();
+            if (!jdkHome.exists() || !jdkHome.isDirectory()) {
+                throw new IOException("Wrong Test JDK Home: " + jdkHome);
+            } else {
+                System.out.println("Using custom Test JDK, home=" + jdkHome);
+            }
+            mconfig.userProperties.put("jvm", new File(jdkHome, "bin/java").getAbsolutePath());
+        }
+        if (StringUtils.isNotBlank(config.getTestJavaArgs())) {
+            mconfig.userProperties.put("argLine", config.getTestJavaArgs());
```

Comment:
I believe this may result in unexpected behaviour if the caller use the `mavenProperties` property to specify an `argLine` for example to bypass to debian openjdk current incompatibility with Surefire...

I believe we should fail the build with a proper informative message if `mavenOptions` contains `argLine` and `testJavaOpts` is used, or merge them or make sure `mavenOptions` are not used for tests runs if `testJavaArgs` is used (and put a message informing of that) or a similar approach.

Reply:
I was just about to comment exactly the same issue. How would be the PCT behave if `mavenOptions` already specify an `argLine`?

Reply:
In such case the setting will be overridden. Maven properties are passed in the line 213, so here we override the setting if passed explicitly. I can add a warning for it if you prefer.

For arg line it would be also possible to append values, but it implies a big risk of collision. I will try this approach and add a warning for the merge.

Reply:
@raul-arabaolaza @fcojfernandez should be addressed now. I have also moved the logic to the upper method so that it is also taken into account in multi-parent hooks

Reply:
LGTM